gammapy/spectrum/energy.py

```diff
@@ -0,0 +1,293 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import print_function, division
+from astropy.units import Quantity
+from astropy.io import fits
+from ..utils.array import array_stats_str
+
+import astropy.units as u
+import numpy as np
+
+__all__ = ['Energy',
+           'EnergyBounds']
+
+class Energy(u.Quantity):
+
+    """Energy bin centers.
+
+    Stored as "ENERGIES" FITS table extensions.
+
+    Parameters
+    ----------
+    energy : `~numpy.array`, scalar, `~astropy.units.Quantity`
+        Energy
+
+    unit : `~astropy.units.UnitBase`, str, optional
+        The unit of the value specified for the energy.  This may be
+        any string that `~astropy.units.Unit` understands, but it is
+        better to give an actual unit object.
+
+    dtype : `~numpy.dtype`, optional
+        See `~astropy.units.Quantity`.
+
+    copy : bool, optional
+        See `~astropy.units.Quantity`.
+
+    """
+
+    def __new__(cls, energy, unit=None, dtype=None, copy=True):
+
+        # Techniques to subclass Quantity taken from astropy.coordinates.Angle
```

Comment:
I think your case is simpler than `Angle` because you do not need to take care of strange input. If it were up to me, I'd just go straight to calling `super(Energy, cls).__new__(cls, ...)` and only after that do the check on whether the unit is that of an energy. That way, it's basically just two lines:
```
        self = super(Energy, cls).__new__(cls, energy, unit,
                                          dtype=dtype, copy=copy)
        if not self.unit.is_equivalent(u.eV):
            raise ...
```